infra/src/main/java/com/redhat/service/bridge/infra/models/filters/FilterFactory.java

```diff
@@ -1,18 +1,31 @@
 package com.redhat.service.bridge.infra.models.filters;
 
+import java.io.IOException;
+import java.util.List;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+
 public class FilterFactory {
-    public static BaseFilter buildFilter(String type, String key, String value) {
-        switch (type) {
-            case StringBeginsWith.FILTER_TYPE_NAME:
-                return new StringBeginsWith(key, value);
-            case StringContains.FILTER_TYPE_NAME:
-                return new StringContains(key, value);
-            case StringEquals.FILTER_TYPE_NAME:
-                return new StringEquals(key, value);
-            case ValuesIn.FILTER_TYPE_NAME:
-                return new ValuesIn(key, value);
-            default:
-                throw new IllegalArgumentException("Type " + type + " is not valid");
+    public static BaseFilter<?> buildFilter(String type, String key, String value) {
+        try {
+            switch (type) {
+                case StringBeginsWith.FILTER_TYPE_NAME:
+                    return new StringBeginsWith(key,
+                            ObjectMapperFactory.get().readValue(value, new TypeReference<List<String>>() {
+                            }));
+                case StringContains.FILTER_TYPE_NAME:
+                    return new StringContains(key, ObjectMapperFactory.get().readValue(value, new TypeReference<List<String>>() {
+                    }));
+                case StringEquals.FILTER_TYPE_NAME:
+                    return new StringEquals(key, value);
+                case ValuesIn.FILTER_TYPE_NAME:
+                    return new ValuesIn(key, ObjectMapperFactory.get().readValue(value, new TypeReference<List<Object>>() {
+                    }));
+                default:
+                    throw new IllegalArgumentException("Type " + type + " is not valid");
+            }
+        } catch (IOException io) {
+            throw new IllegalArgumentException("Value " + value + " is not valid");
```

Comment:
@fjtirado Can you please chain the `IOException` in the constructor so that we don't lose the full stracktrace.